src/NHibernate/Cache/NonstrictReadWriteCache.cs

```diff
@@ -261,5 +274,11 @@ public bool AfterInsert(CacheKey key, object value, object version)
 		{
 			return false;
 		}
+
+		private void CheckCache()
+		{
+			if (_cache == null || _isDestroyed)
+				throw new InvalidOperationException(_isDestroyed ? "The cache has already been destroyed" : "The concrete cache is not defined");
```

Comment:
These strategies classes are instantiated with a `null` underlying cache, and a later set on their `Cache` property is supplying their underlying cache. Nothing prevents this underlying cache of staying `null` or being set to `null`, without the cache having been destroyed.

So we cannot just "guess" that the strategy has been destroyed if its underlying cache is `null`, it may be wrong.

I do not see any valid reason for the build pattern actually used by theses classes. It should likely be reworked for taking the underlying cache as a constructor parameter, mandatory, and setters should be obsoleted and removed in next major.
But that would be out of the scope of this PR.